carp.common/src/commonMain/kotlin/dk/cachet/carp/common/FullyQualifiedName.kt

```diff
@@ -0,0 +1,74 @@
+package dk.cachet.carp.common
+
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.descriptors.PrimitiveKind
+import kotlinx.serialization.descriptors.PrimitiveSerialDescriptor
+import kotlinx.serialization.descriptors.SerialDescriptor
+import kotlinx.serialization.encoding.Decoder
+import kotlinx.serialization.encoding.Encoder
+
+
+/**
+ * A uniquely named entity within a [namespace].
+ * TODO: This likely needs more restrictions in line with other standards. Disallow certain symbols? Lower-case only?
+ */
+@Serializable( with = FullyQualifiedNameSerializer::class )
+data class FullyQualifiedName(
```

Comment:
I don't think this is as general as you imply. It's only used for data types. If you want to use the same validation elsewhere, then move it out into an `FQNValidator` or something. DataType becomes an unnecessary wrapper around a generalization that, well, is its own identity.

Reply:
I added this because the next feature I'll add likely also needs FQN.

> I added this in anticipation of needing some universal key registration for participant metadata, which would also benefit from fully qualified names.

A `FQNValidator` does sound like a likely cleaner solution!

Reply:
In essence, `DataType` _is_ a FQN. I don't see it deviating from it. Anything that might deviate from it is meta-data attached to that FQN. 🤔

Reply:
Related:
- https://github.com/cph-cachet/carp.core-kotlin/issues/172
- https://github.com/cph-cachet/carp.core-kotlin/issues/132